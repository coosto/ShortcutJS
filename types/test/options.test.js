"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const options_1 = require("../src/options");
describe('Options', () => {
    it('has expected defaults if no params are passed', () => {
        expect(new options_1.Options()).toEqual({ debug: false, preventDefault: false, onlyStateCombos: false });
    });
    it('merges options when valid options are passed', () => {
        const options = new options_1.Options();
        options.debug = true;
        expect(new options_1.Options({ debug: true })).toEqual(options);
    });
    it('throws error if a non-expected property is passed', () => {
        expect(() => new options_1.Options({ ppe: 3 })).toThrowErrorMatchingSnapshot();
    });
});
