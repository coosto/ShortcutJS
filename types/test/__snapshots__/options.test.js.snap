exports[`Options throws error if a non-expected property is passed 1`] = `"Some of the options are not correct. Checkout the docs at https://github.com/coosto/ShortcutJS for more info"`;
